tests/pageobjects/classes.po.ts

```diff
@@ -1,9 +1,72 @@
-import { Selector } from 'testcafe';
+import { Selector, t } from 'testcafe';
 
 export default class ClassesPage {
   static url = '/class';
   _id: Selector;
+  scheduleTestClassNameCell: Selector;
+  scheduleTestClassDeleteButton: Selector;
+  confirmDeleteButton: Selector;
+  newClassButton: Selector;
+  className: Selector;
+  classGradeDropDown: Selector;
+  classGradeFirstOption: Selector;
+  allClassesButton: Selector;
+  scheduleCell: Selector;
+  editCellDialogue: Selector;
+  editCellLesson: Selector;
+  lessonOption: Selector;
+  editCellLocation: Selector;
+  locationOption: Selector;
+  editCellUpdateButton: Selector;
+  editCellCloseButton: Selector;
+
   constructor() {
     this._id = Selector('[data-test-id$="classes-page"]');
+    this.scheduleTestClassNameCell = Selector('.class-name').withExactText('scheduleTestClass');
+    this.scheduleTestClassDeleteButton = Selector('[data-test-id$="delete-class-button-scheduleTestClass"]');
+    this.confirmDeleteButton = Selector('[id$="confirm-delete-class"]');
+    this.newClassButton = Selector('.mat-button-wrapper').withExactText('הוספת כיתה חדשה');
+    this.className = Selector('[id$="class-name"]');
+    this.classGradeDropDown = Selector('[id$="class-grade"]');
+    this.classGradeFirstOption = Selector('.mat-option-text').withExactText('א');
+    this.allClassesButton = Selector('.back');
+    this.scheduleCell = Selector('[role$="gridcell"]').nth(10);
+    this.editCellDialogue = Selector('[data-test-id$="edit-cell-dialogue"]');
+    this.editCellLesson = Selector('[data-test-id$="lessons-dropdown"]');
+    this.lessonOption = Selector('.lessons-option');
+    this.editCellLocation = Selector('[data-test-id$="locations-dropdown"]');
+    this.locationOption = Selector('.locations-option');
+    this.editCellUpdateButton = Selector('[data-test-id$="update-edit-lesson-dialogue"]');
+    this.editCellCloseButton = Selector('[data-test-id$="close-edit-lesson-dialogue"]');
+  }
+
+  async createNewScheduleTestClass() {
```

Comment:
from my understanding, `po` files should describe the page, essentially expose `Selector` instances. Actions like `click` etc should be placed in the `spec` file itself as they are part of the "when" phase of the test.

Reply:
let me explain the thought behind this. this practice is taken from other test frameworks - where the page objects include common Actions on a page. so that any test can reuse the specific Action code (class creation in this case). the spec files mainly specify the actual Assertions. the point is that common atomic actions are placed in the PO for reuse. what are your thoughts about this?

Reply:
I understand the approach but could argue from my own experience that this makes the test less expressive and creating a "black box" code that does "magic". 

The code that I would take to a separate file is the mocking initiation (if needed) all the rest in my opinion should be inside the tests itself to explicitly show the 
```
// given
// when
// then
```
steps.

Reply:
Done